manager/integration/tests/test_migration.py

```diff
@@ -147,7 +146,55 @@ def test_migration_with_failed_replica(clients, volume_name):  # NOQA
        And the old failed replica will be cleaned up.
     8. Validate initially written test data.
     """
-    return
+    def finalizer():
+        exec_cmd = ["mkdir", "/var/lib/longhorn/replicas"]
+        subprocess.check_output(exec_cmd)
+
+    request.addfinalizer(finalizer)
+
+    client, volume, data = setup_migration_test(clients,
+                                                volume_name,
+                                                replica_cnt=3)
+
+    current_node = common.get_self_host_id()
+    hosts = get_hosts_for_migration_test(clients)
+    migrate_target = hosts[0]
+    volume.attach(hostId=current_node)
+    volume = common.wait_for_volume_healthy(client, volume_name)
+
+    old_replicas = []
+    for replica in volume.replicas:
+        old_replicas.append(replica.name)
+
+    exec_cmd = ["rm", "-rf",  "/var/lib/longhorn/replicas"]
+    subprocess.check_output(exec_cmd)
+    volume = common.wait_for_volume_degraded(client, volume_name)
+
+    volume.attach(hostId=migrate_target)
+    volume = common.wait_for_volume_migration_ready(client, volume_name)
+
+    new_replicas = 0
+    for replica in volume.replicas:
+        if replica.hostId == current_node:
+            assert replica.running is False
+        else:
+            assert replica.running is True
+            if replica.name not in old_replicas:
+                new_replicas = new_replicas + 1
+    assert new_replicas == 2
+
+    volume.detach(hostId=current_node)
+
+    volume = common.wait_for_volume_migration_node(client,
+                                                   volume_name,
+                                                   migrate_target)
+
+    volume.updateReplicaCount(replicaCount=2)
```

Comment:
A little confusing, so after migration, the number of replica will not be back to 3? Why we need to specially update the replica count to 2?

Reply:
This verifies that the old/un-migrated replica is still removable.